pennylane/templates/layers/quantum_number_preserving_u2.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the QuantumNumberPreserving Gate Fabric template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+import numpy as np
+import pennylane as qml
+from pennylane.operation import Operation, AnyWires
+
+
+class QuantumNumberPreservingU2(Operation):
+    r"""Implements a local, expressive, and quantum-number-preserving ansatz using VQE circuit fabrics
+    proposed by Anselmetti *et al.* in `arXiv:2104.05692 <https://arxiv.org/abs/2104.05692>`_.
+
+    This template prepares the :math:`2M` qubits trial state, where `M` is the number of spatial orbitals.
+    It uses 4-local-nearest-neighbor-tessellation of alternating even and off spatial-orbitals-pair 2-parameter,
```

Comment:
even and odd? I find this sentence a bit hard to read.

Reply:
I have rewritten the entire thing to make it sound crisper.